modules/openapi-generator/src/main/resources/dart2/class.mustache

```diff
@@ -190,22 +188,22 @@ class {{{classname}}} {
   static List<{{{classname}}}> listFromJson(List<dynamic> json, {bool emptyIsNull, bool growable,}) =>
     json == null || json.isEmpty
       ? true == emptyIsNull ? null : <{{{classname}}}>[]
-      : json.map((v) => {{{classname}}}.fromJson(v)).toList(growable: true == growable);
+      : (json as List<Map<String, dynamic>>).map({{{classname}}}.fromJson).toList(growable: true == growable);
 
   static Map<String, {{{classname}}}> mapFromJson(Map<String, dynamic> json) {
     final map = <String, {{{classname}}}>{};
-    if (json != null && json.isNotEmpty) {
-      json.forEach((String key, dynamic v) => map[key] = {{{classname}}}.fromJson(v));
+    if (json?.isNotEmpty == true) {
+      json.forEach((key, value) => map[key] = {{{classname}}}.fromJson(value as Map<String, dynamic>));
     }
     return map;
   }
 
   // maps a json object with a list of {{{classname}}}-objects as value to a dart map
   static Map<String, List<{{{classname}}}>> mapListFromJson(Map<String, dynamic> json, {bool emptyIsNull, bool growable,}) {
     final map = <String, List<{{{classname}}}>>{};
-    if (json != null && json.isNotEmpty) {
-      json.forEach((String key, dynamic v) {
-        map[key] = {{{classname}}}.listFromJson(v, emptyIsNull: emptyIsNull, growable: growable);
+    if (json?.isNotEmpty == true) {
```

Comment:
do we need `== true`?

Reply:
yes

```dart
void main() {
  if(null) {
    print("x");
  }
}

// Uncaught Error: Assertion failed: "boolean expression must not be null"
```

Reply:
I understand this change but in any case we don't care about wether it is empty. `foreach` doesn't run if it is empty.
Maybe something like this but thats just my preference, up  to you.
```dart
return json?.map((key, dynamic value) => MapEntry(key, {{{classname}}}.listFromJson(value, emptyIsNull: emptyIsNull, growable: growable,),) ?? <String, {{{classname}}}>{};
```

Reply:
Ideally you could change this in another PR as it is probably related to some analyzer rule.